### PR TITLE
Extended patient create: Sex and birth

### PR DESCRIPTION
**Pull request by mpeels**
## Description

Sex and birth section.

## Tickets

* [CNFT1-2777](https://cdc-nbs.atlassian.net/browse/CNFT1-2777)

## Demo
birth sex

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

[CNFT1-2777]: https://cdc-nbs.atlassian.net/browse/CNFT1-2777?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-ui/src/date/asAgeDisplay.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { differenceInDays, differenceInMonths, differenceInYears } from 'date-fns';
+
+const asAgeDisplay = (dateOfBirth: string | Date | null | undefined, from: Date = new Date()): string | undefined => {
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/date/asAgeDisplay.ts:3]: suggestion (non-blocking): There are some other functions( [displayName](https://github.com/CDCgov/NEDSS-Modernization/blob/main/apps/modernization-ui/src/name/displayName.ts) and [displayAddress](https://github.com/CDCgov/NEDSS-Modernization/blob/main/apps/modernization-ui/src/address/display/displayAddress.ts)) for displaying things that are named `displayX` should we keep that pattern?  

nit: I may have been corrupted by too much functional programming but if you move the parameters around and add some currying it becomes a little more expressive.

`const asAgeDisplay = (from: Date = new Date()) = (dateOfBirth: Maybe<string | Date>) => {...}`. you could then write functions from this like `const asAgeFromToday = asAgeDisplay();`

File: apps/modernization-ui/src/apps/patient/data/sexAndBirth/asSex.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { asValue } from 'options';
-import { Sex } from './api';
-import { SexEntry } from './entry';
+import { Sex } from '../api';
+import { SexEntry } from '../entry';
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/patient/data/sexAndBirth/asSex.ts:3]: question (non-blocking): Should the types be moved down a folder also?

File: apps/modernization-ui/src/apps/patient/add/extended/nav/AddPatientExtendedNav.tsx [hunk 3]
```diff
@@ -13,7 +13,7 @@ export const AddPatientExtendedNav = () => {
                     <a href="#identification">Identification</a>
                     <a href="#section-Race">Race</a>
                     <a href="#ethnicity">Ethnicity</a>
-                    <a href="#section-SexAndBirth">Sex & birth</a>
+                    <a href="#sexAndBirth">Sex & birth</a>
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/patient/add/extended/nav/AddPatientExtendedNav.tsx:16]: question: `sex-and-birth` or `sexAndBirth`?  For some reason kebab case looks better in urls

**Review by adamloup-enquizit — APPROVED** [review]
👍 Looks good, added a few non-blocking comments

File: apps/modernization-ui/src/apps/patient/add/extended/nav/AddPatientExtendedNav.tsx (hunk 3 above)
Comment (mpeels) [reply apps/modernization-ui/src/apps/patient/add/extended/nav/AddPatientExtendedNav.tsx:16]: agreed. Will see if Henry will update those as part of his nav component PR

File: apps/modernization-ui/src/date/asAgeDisplay.ts (hunk 1 above)
Comment (mpeels) [reply apps/modernization-ui/src/date/asAgeDisplay.ts:3]: Will add this as part of Mortality PR

File: apps/modernization-ui/src/apps/patient/data/sexAndBirth/asSex.ts (hunk 2 above)
Comment (mpeels) [reply apps/modernization-ui/src/apps/patient/data/sexAndBirth/asSex.ts:3]: Maybe. That would get rid of `api` and `entry`

**Pull request merged**